File: 0201-0300/0201.Bitwise_AND_of_Numbers_Range.java
//3ms





class Solution { 
    public int rangeBitwiseAnd(int m, int n) { 
        int i = 0; 
        for (; m != n; ++i) { 
            m >>= 1; 
            n >>= 1; 
        } 
        return n << i; 
    } 
}     






//3ms






class Solution {
    public int rangeBitwiseAnd(int left, int right) {
            while(left<right){
            right = right & (right-1);
        }
        return right;
    }
}

File: 0201-0300/0201.Bitwise_AND_of_Numbers_Range.py
#3ms





class Solution:
    def rangeBitwiseAnd(self, left: int, right: int) -> int:
        while right>left:
            right &=(right-1)
        return right
        




#0ms







class Solution:
    def rangeBitwiseAnd(self, left: int, right: int) -> int:
        return left & (-1 << (left ^ right).bit_length())






#3ms






class Solution:
    def rangeBitwiseAnd(self, left: int, right: int) -> int:
        shift = 0
        
        while left < right:
            left >>= 1
            right >>= 1
            shift += 1
        
        return left << shift
